### PR TITLE
Access: Fetch fresh permissions for target GlobalOrgID in AuthorizeInOrgMiddleware

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Ensure fresh permissions are loaded for AuthorizeInOrg requests

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/accesscontrol/authorize_in_org_test.go [hunk 1]
```diff
@@ -70,6 +72,30 @@ func TestAuthorizeInOrgMiddleware(t *testing.T) {
...
 		},
+		{
+			name: "should return 200 when user has permissions for a global org",
+			orgIDGetter: func(c *contextmodel.ReqContext) (int64, error) {
+				return accesscontrol.GlobalOrgID, nil
+			},
+			evaluator:       accesscontrol.EvalPermission("users:read", "users:*"),
+			accessControl:   ac,
+			userCache:       &usertest.FakeUserService{},
+			ctxSignedInUser: &user.SignedInUser{UserID: 1, OrgID: 1, Permissions: map[int64]map[string][]string{0: {"users:read": {"users:*"}}}},
```
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/authorize_in_org_test.go:83]: nit: can we do this to make the test a bit more readable:
```suggestion
			ctxSignedInUser: &user.SignedInUser{UserID: 1, OrgID: 1, Permissions: map[int64]map[string][]string{accesscontrol.GlobalOrgID: {"users:read": {"users:*"}}}},
```

**Review by IevaVasiljeva — APPROVED** [review]
LGTM!

File: pkg/services/accesscontrol/middleware.go [hunk 2]
```diff
@@ -190,19 +190,24 @@ func AuthorizeInOrgMiddleware(ac AccessControl, service Service, cache userCache
...
-			if targetOrgID != GlobalOrgID && userCopy.Permissions[targetOrgID] == nil {
-				query := user.GetSignedInUserQuery{UserID: c.UserID, OrgID: targetOrgID}
-				queryResult, err := cache.GetSignedInUserWithCacheCtx(c.Req.Context(), &query)
-				if err != nil {
-					deny(c, nil, fmt.Errorf("failed to authenticate user in target org: %w", err))
-					return
+			if userCopy.OrgID != targetOrgID {
+				switch targetOrgID {
+				case GlobalOrgID:
+					userCopy.OrgID = GlobalOrgID
```
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/middleware.go:196]: Hmm, actually, do we need to clear the org role here? In general, would it be safer to create a new signed in user struct from scratch?

Comment (Jguer) [reply pkg/services/accesscontrol/middleware.go:196]: yes, but we might lose some user context. Partially afraid.

**Pull request merged**